### PR TITLE
mgr/dashboard: add API team to CODEOWNERS

### PR DESCRIPTION
**Pull request by epuertat**
Fixes: https://tracker.ceph.com/issues/45705
Signed-off-by: Ernesto Puerta <epuertat@redhat.com>

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by s0nea** [comment]
Hi @epuertat does this also apply to the QA API tests, e.g. `/qa/tasks/mgr/dashboard`?

**Comment by epuertat** [comment]
> Hi @epuertat does this also apply to the QA API tests, e.g. `/qa/tasks/mgr/dashboard`?

Good catch @s0nea, thanks! I completely forgot about the teuthology stuff! Will update this.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -7,6 +7,14 @@
...
 /doc/mgr/dashboard.rst                          @ceph/dashboard
 
+# Dashboard API team
+/src/pybind/mgr/dashboard/controllers                  @ceph/api
+/src/pybind/mgr/dashboard/frontend/src/app/shared/api  @ceph/api
+/src/pybind/mgr/dashboard/run-backend-api-tests.sh     @ceph/api
+/qa/tasks/mgr/dashboard                                @ceph/api
+/qa/tasks/mgr/test_module_selftest.py                  @ceph/api
+/qa/tasks/mgr/test_dashboard.py                        @ceph/api
+
```
Comment (s0nea) [inline .github/CODEOWNERS:17]: Thank you for adding these!

I'm wondering if `/qa/suites/rados/dashboard` or `/qa/suites/rados/dashboard/tasks/dashboard.yaml` (I'm not really sure which of these two makes more sense) can be added here as well.

Comment (epuertat) [reply .github/CODEOWNERS:18]: Thanks! Updated with the first suggestion.

Comment (s0nea) [reply .github/CODEOWNERS:18]: Thank you!

**Pull request merged**